fix(getters): Do not use store.jwt to compute hasLogIn

arthurdenner opened the pull request:
<!--
Please make sure you are familiar with and follow the instructions in the
contributing guidelines (found in the CONTRIBUTING.md file).

Please fill out the information below to expedite the review and (hopefully)
merge of your pull request!
-->

<!-- Is it a Bug fix, feature, docs update, ... -->
**What kind of change does this PR introduce?**

This PR aims to fix two bugs I noticed related to sign out that are described below.

<!-- You can also link to an open issue here -->
**What is the current behavior?**
- When you're on the dashboard and signs out, the dashboard doesn't update;
- When you're on a sandbox page and signs out, the sidebar doesn't update;

I believe the problem it's in how `hasLogIn` is computed, because it uses `store.jwt`, which doesn't get updated in the `signOut` sequence. I'm not sure if removing it from the computation is the proper fix to these problems, but since in [this commit](https://github.com/codesandbox/codesandbox-client/commit/705679b88d6057583587b8ab32539f3062a4a479#diff-fcc68a385e98ecb8fed8d41e68d69c39L14) the logic was changed to add `!!this.jwt` and not only use `!!store.get('jwt')` I thought that maybe it is.

<!-- if this is a feature change -->
**What is the new behavior?**

When you sign out in the situations described above, the UI is updated as expected.

<!-- Have you done all of these things?  -->
**Checklist**:
<!-- add "N/A" to the end of each line that's irrelevant to your changes -->
<!-- to check an item, place an "x" in the box like so: "- [x] Documentation" -->
- [ ] Documentation
- [ ] Tests
- [ ] Ready to be merged <!-- In your opinion, is this ready to be merged as soon as it's reviewed? -->
- [ ] Added myself to contributors table <!-- this is optional, see the contributing guidelines for instructions -->

<!-- feel free to add additional comments -->

<!-- Thank you for contributing! -->

arthurdenner commented:
@SaraVieira @CompuIves pinging you to see if this PR can get a review

SaraVieira commented:
Hey!

Sorry for the delay

Can you please explain the change?

Im also pinging @CompuIves

arthurdenner commented:
Hi Sara! :wave:  No worries.

Try to reproduce what I described in the current behavior section and you'll see that the UI doesn't update as expected.

When I was debugging it, I found that removing the `store.jwt` fix the problem and (if I'm not mistaken) wherever `hasLogin` is being used continues to behave as before.

Sorry I'm on the phone right now and can't provide more information, but if you need any other details I can provide later today.

arthurdenner commented:
Current problems:

![csb-logout-dashboard](https://user-images.githubusercontent.com/13774309/58484451-786be780-8138-11e9-9c0c-dfe530fcdaef.gif)
![csb-logout-sandbox](https://user-images.githubusercontent.com/13774309/58484452-786be780-8138-11e9-875b-6b2f1838d576.gif)

lbogdan commented:
@arthurdenner Can you please rebase this to `master`, so that it gets built by the new PR build system? Thanks!

arthurdenner commented:
@lbogdan, rebased.

I also updated my previous comment to add the new behaviour after the PR.

To reinforce: I'm not sure if this PR contains the proper fix to these problems because I didn't tried to logout during a live session, but the remaining scenarios where `hasLogin` is being used are working as before.

arthurdenner commented:
@lbogdan, is there any way I see exactly why the integration tests are failing and maybe fix them? :thinking:

CompuIves commented:
This is great @arthurdenner!! Just tested this and the logic is far better now. Thank you!
